src/systems.rs

```diff
@@ -120,12 +120,16 @@ pub fn particle_spawner(
             continue;
         }
 
-        for _ in 0..to_spawn + extra {
-            let origin_pos = match particle_system.space {
-                ParticleSpace::Local => Transform::default(),
-                ParticleSpace::World => Transform::from(*global_transform),
-            };
+        let origin_pos = match particle_system.space {
+            ParticleSpace::Local => Transform::default(),
+            ParticleSpace::World => Transform::from(*global_transform),
+        };
+        let origin_z = match particle_system.space {
+            ParticleSpace::Local => transform.translation.z,
```

Comment:
Shouldn't this just be 0? Since the particle is parented under the particle system? Otherwise, we'd end up at a z value of twice the origin_z?

Reply:
🤦 Yup, you're right.